### PR TITLE
Add theme toggle

### PR DESCRIPTION
**Pull request by neslihanturan**
Night mode theme check box is added to Settings Activity related with #428

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/SettingsActivity.java [hunk 1]
```diff
@@ -68,7 +79,12 @@ protected void onPause() {
 
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-
+        if(key.equals("theme")){
+            //Finish current activity and tart new one with selected theme
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/SettingsActivity.java]: Typo in word "start"

Also kinda related - maybe just a small thing but using spaces between slashes and comment text, as per [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace)

File: app/src/main/java/fr/free/nrw/commons/SettingsActivity.java [hunk 2]
```diff
@@ -21,6 +25,13 @@
     protected void onCreate(Bundle savedInstanceState) {
         getDelegate().installViewFactory();
         getDelegate().onCreate(savedInstanceState);
+        //Check prefs on every activity starts
+        if (PreferenceManager.getDefaultSharedPreferences(this).getBoolean("theme",false)) {
+            setTheme(R.style.DarkAppTheme);
+        }else {
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/SettingsActivity.java]: Spacing

File: app/src/main/java/fr/free/nrw/commons/theme/BaseActivity.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+public class BaseActivity extends Activity {
+    boolean currentTheme;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        if (PreferenceManager.getDefaultSharedPreferences(this).getBoolean("theme",false)) {
+            currentTheme = true;
+            setTheme(R.style.DarkAppTheme);
+        }else {
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/theme/BaseActivity.java]: Spacing

File: app/src/main/java/fr/free/nrw/commons/theme/BaseAppCompatActivity.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+public class BaseAppCompatActivity extends AppCompatActivity {
+    boolean currentTheme;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        if (PreferenceManager.getDefaultSharedPreferences(this).getBoolean("theme",false)) {
+            currentTheme = true;
+            setTheme(R.style.DarkAppTheme);
+        }else {
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/theme/BaseAppCompatActivity.java]: Spacing

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -33,6 +33,7 @@
   <string name="menu_from_camera">Take photo</string>
   <string name="menu_nearby">Nearby</string>
   <string name="provider_contributions">My uploads</string>
+  <string name="menu_theme_toggle">Night mode</string>
```
Comment (domdomegg) [inline app/src/main/res/values/strings.xml]: Not required anymore

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -91,6 +92,8 @@ Tap this message (or hit back) to skip this step.</string>
   <string name="use_previous">Use previous title/description</string>
   <string name="allow_gps">Automatically get current location</string>
   <string name="allow_gps_summary">Retrieve current location to offer category suggestions if image is not geotagged</string>
+  <string name="preference_theme">Select theme</string>
```
Comment (domdomegg) [inline app/src/main/res/values/strings.xml]: Maybe a strings would be:
preference_theme: "Night mode"
preference_theme_summary: "Use dark theme"

File: app/src/main/res/layout/fragment_media_detail.xml [hunk 7]
```diff
@@ -3,7 +3,7 @@
     android:orientation="vertical"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@android:color/background_dark"
+    android:background="@android:color/background_light"
```
Comment (domdomegg) [inline app/src/main/res/layout/fragment_media_detail.xml]: Use `?attr/mainBackground` instead
https://github.com/neslihanturan/apps-android-commons/pull/3

**Comment by misaochan** [comment]
Great work @neslihanturan ! :) Works well for me in a brief test, so I'm happy for this to be merged after @domdomegg 's suggestions have been implemented.

**Comment by whym** [comment]
I'm tweaking some of the Codacy settings. In the meantime, some not-too-important warnings might be sent here. Sorry about that.

**Comment by domdomegg** [comment]
I'm happy to merge if we assume Codacy is being overzealous, and we can fix code style/lint issues later.

**Comment by misaochan** [comment]
Awesome, thanks everyone!

**Pull request merged**